app/supplier_utils.py

```diff
@@ -9,11 +9,7 @@ def validate_and_return_supplier_request(supplier_id=None):
     json_has_required_keys(json_payload['suppliers'], ['contactInformation'])
 
     # remove unnecessary fields
-    json_payload['suppliers'] = drop_foreign_fields(json_payload['suppliers'], ['links'])
-    json_payload['suppliers']['contactInformation'] = [
-        drop_foreign_fields(contact_data, ['links'])
-        for contact_data in json_payload['suppliers']['contactInformation']
```

Comment:
I don't think so, because the crash is when 'contactInformation' is there, but is not a list (or, worse, it's some kind of iterable that doesn't contain a dictionary)